### PR TITLE
Bump solana_rbpf to v0.7.0

### PR DESCRIPTION
**Pull request by Lichtso**
#### Problem
[solana_rbpf v0.7.0](https://github.com/solana-labs/rbpf/releases/tag/v0.7.0) was released.

#### Summary of Changes
- Removes the generic parameter `Verifier` from `Executable`.
- Uses `FunctionRegistry` for `BuiltinProgram` as well.
- Generates and manages the `runtime_environment_key` inside RBPF again.
- Updates the "test_elfs" to the newest SBPFv2 ISA.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -294,7 +294,7 @@ siphasher = "0.3.11"
 smpl_jwt = "0.7.1"
 socket2 = "0.5.3"
 soketto = "0.7"
-solana_rbpf = "=0.6.0"
+solana_rbpf = { git = "https://github.com/solana-labs/rbpf", branch = "main" }
```
Comment (dmakarov) [inline Cargo.toml:297]: Is this how this PR will be merged, or you're planning to change this to a specific version (0.7.0)?

Comment (Lichtso) [reply Cargo.toml:297]: No it is not, just for CI before I publish the RBPF release. Thus, the commit is marked as work-in-progress (WIP).

File: programs/bpf_loader/src/lib.rs [hunk 2]
```diff
@@ -498,12 +494,10 @@ fn process_instruction_inner(
...
         .find_program_in_cache(program_account.get_key())
-        .ok_or(InstructionError::InvalidAccountData)?;
-
-    if executor.is_tombstone() {
-        return Err(Box::new(InstructionError::InvalidAccountData));
-    }
-
+        .ok_or_else(|| {
+            ic_logger_msg!(log_collector, "Program is not cached");
+            InstructionError::InvalidAccountData
```
Comment (alessandrod) [inline programs/bpf_loader/src/lib.rs:499]: Why is the is_tombstone check not needed anymore here?

Comment (Lichtso) [reply programs/bpf_loader/src/lib.rs:499]: It is redundant. See implementation of `is_tombstone()` [here](https://github.com/solana-labs/solana/blob/2f7a76760e365f76c0a16745ff87cdd77e706b6e/program-runtime/src/loaded_programs.rs#L330), which matches exactly the three cases below and also throws `InstructionError::InvalidAccountData`:
```
LoadedProgramType::FailedVerification(_)
| LoadedProgramType::Closed
| LoadedProgramType::DelayVisibility
```

That has nothing to do with the RBPF bump, so I can split it into a separate PR.

Comment (alessandrod) [reply programs/bpf_loader/src/lib.rs:499]: I see, yes please split it out if you don't mind

Comment (Lichtso) [reply programs/bpf_loader/src/lib.rs]: See #33145

**Review by alessandrod — APPROVED** [review]

**Pull request merged**